sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -46,17 +45,18 @@ object JDBCRDD extends Logging {
    * Takes a (schema, table) specification and returns the table's Catalyst
    * schema.
    *
-   * @param url - The JDBC url to fetch information from.
-   * @param table - The table name of the desired table.  This may also be a
-   *   SQL query wrapped in parentheses.
+   * @param options - JDBC options that contains url, table and other information.
    *
    * @return A StructType giving the table's Catalyst schema.
    * @throws SQLException if the table specification is garbage.
    * @throws SQLException if the table contains an unsupported type.
    */
-  def resolveTable(url: String, table: String, properties: Properties): StructType = {
+  def resolveTable(options: JDBCOptions): StructType = {
+    val url = options.url
+    val table = options.table
+    val properties = options.asProperties
```

Comment:
Yes, when we use `write.formt("jsbd").option` api, they will be set[1]. We are already setting `fetchsize`, `batchsize` and `isolattionlevel` here which seem also hanlded within Spark. Do you see any potential problem from this?

[1]https://github.com/apache/spark/pull/15292/files/23e46045d2a2a11eb0f0f50762216242b2f5c633#diff-112a12b90840da372611b544efceef51L49